File: majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int majority = nums[0];
        int count = 1;

        for(int i = 1; i < nums.length; i++) {
            int curNum = nums[i];
            
            if(count == 0) {
                majority = curNum;
                count = 1;
                continue;
            }

            if(majority == curNum) count++;
            else count--;
        }

        return majority;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        majority = 1
        element = nums[0]

        for i in range(1, len(nums)):
            if nums[i] == element:
                majority += 1
            else:
                majority -= 1
            
            if majority == 0:
                element = nums[i]
                majority = 1
        
        return element